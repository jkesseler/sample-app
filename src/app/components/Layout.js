import React, { memo } from 'react';
import PropTypes from 'prop-types';

import {
  AppBar, Paper, Toolbar, Typography,
} from '@material-ui/core';

export const Layout = ({ children }) => (
  <Paper
    elevation={0}
    style={{
      padding: 0,
      margin: 0,
      backgroundColor: '#fafafa',
    }}
  >
    <AppBar color="primary" position="static" style={{ height: 64 }}>
      <Toolbar style={{ height: 60 }}>
        <Typography color="inherit">Todo App</Typography>
      </Toolbar>
    </AppBar>
    { children }
  </Paper>
);

Layout.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default memo(Layout);
