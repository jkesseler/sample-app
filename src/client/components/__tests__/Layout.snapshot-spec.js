import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
// non memoized component
import { Layout } from '../Layout';

const Child = () => (<span>fake child</span>);

describe('Layout', () => {
  it('renders without crashing', () => {
    const component = shallow(<Layout><Child /></Layout>);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
