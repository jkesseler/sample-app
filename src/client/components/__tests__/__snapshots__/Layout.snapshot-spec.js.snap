// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders without crashing 1`] = `
<WithStyles(Paper)
  elevation={0}
  style={
    Object {
      "backgroundColor": "#fafafa",
      "margin": 0,
      "padding": 0,
    }
  }
>
  <WithStyles(AppBar)
    color="primary"
    position="static"
    style={
      Object {
        "height": 64,
      }
    }
  >
    <WithStyles(Toolbar)
      style={
        Object {
          "height": 60,
        }
      }
    >
      <WithStyles(Typography)
        color="inherit"
      >
        Todo App
      </WithStyles(Typography)>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
  <Child />
</WithStyles(Paper)>
`;
